### PR TITLE
Fix epoch processing

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Fixes problems that arise during epoch processing due to not updating block roots properly during slot processing.

### CODE REVIEW

File: services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainService.java [hunk 1]
```diff
@@ -56,7 +56,8 @@ public void afterChainStart(Boolean chainStarted) {
       scheduler.scheduleAtFixedRate(
           new SlotScheduler(this.eventBus),
           initialDelay,
-          Constants.SLOT_DURATION,
+          //Constants.SLOT_DURATION,
+          6,
```
Comment (jrhea) [inline services/beaconchain/src/main/java/tech/pegasys/artemis/services/beaconchain/BeaconChainService.java:59]: This change should probably be reverted

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java [hunk 2]
```diff
@@ -15,45 +15,28 @@
...
-    // TODO: change values to UnsignedLong
-    int currSlot = state.getSlot().intValue();
-    List<Bytes32> latestRandaoMixes = state.getLatest_randao_mixes();
-    int index = (currSlot - 1) % LATEST_RANDAO_MIXES_LENGTH;
-    Bytes32 prevSlotRandaoMix = latestRandaoMixes.get(index);
-    latestRandaoMixes.set(currSlot % LATEST_RANDAO_MIXES_LENGTH, prevSlotRandaoMix);
-  }
 
-  public static void updateRecentBlockHashes(BeaconState state, BeaconBlock block)
+  public static void updateBlockRoots(BeaconState state, Bytes32 previous_block_root, String logPrefix)
```
Comment (jrhea) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java]: I believe that this method is superior to what is currently in master bc you are referencing the real previous_block.

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java [hunk 3]
```diff
@@ -15,45 +15,28 @@
...
 import net.consensys.cava.bytes.Bytes32;
 import tech.pegasys.artemis.datastructures.Constants;
-import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
 import tech.pegasys.artemis.datastructures.state.BeaconState;
 import tech.pegasys.artemis.datastructures.util.BeaconStateUtil;
 import tech.pegasys.artemis.statetransition.StateTransitionException;
 
 public class SlotProcessorUtil {
-  public static void updateLatestRandaoMixes(BeaconState state) {
-    // TODO: change values to UnsignedLong
```
Comment (jrhea) [inline ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java:33]: Why did this go away?  Was this intentional?  If so, I get this error after running: `./gradlew mash`

```
> Task :ethereum:statetransition:compileTestJava FAILED
/Users/jonny/projects/consensys/pegasys/artemis/ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/StateTransitionTest.java:74: error: 
cannot find symbol
    SlotProcessorUtil.updateLatestRandaoMixes(state);
                     ^
  symbol:   method updateLatestRandaoMixes(BeaconState)
  location: class SlotProcessorUtil
1 error
```

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconState.java [hunk 4]
```diff
@@ -29,42 +29,42 @@
 
 public class BeaconState {
   // Misc
-  private UnsignedLong slot;
-  private UnsignedLong genesis_time;
-  private Fork fork; // For versioning hard forks
+  protected UnsignedLong slot;
```
Comment (jrhea) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconState.java]: If you aren't making a new class that inherits from BeaconState stores previousBlock, then we should probably just leave these member variables as is.

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconStateWrapper.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package tech.pegasys.artemis.datastructures.state;
+
+public class BeaconStateWrapper extends BeaconState
+{
```
Comment (jrhea) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconStateWrapper.java]: You don't need this anymore, right?

File: ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java (hunk 3 above)
Comment (cemozerr) [reply ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/SlotProcessorUtil.java:33]: Yes intentional. It's not in the spec anymore.

**Review by jrhea — APPROVED** [review]
less lines of code + verified each line of rebase + runs faster = lgtm

**Pull request merged**